### DISCUSSION TITLE
Multitargeting .NetStandard and netx.0

### DISCUSSION THREAD

**Discussion by JKorf** [dotnet/runtime#110689] [Q&A] [answered]
Hi, as a library developer currently targeting .Net standard 2.0/2.1 for max compatibility, is there any benefit to also multi target different dotnet core versions so System.Text.Json could potentially use functionality only available in newer dotnet versions?

Current setup:
`[User application - anywhere from .net Framework to dotnet 9.0]` => `[My library - .net standard 2.0/2.1]` => `[System.Text.Json - 9.0.0]`

**Comment by huoyaoyuan** [comment, marked as answer]
You don't need to pay additional effort for *transitive* features. The implementations of libraries are picked based on entry application. The .NET 9 version of System.Text.Json will always be used for .NET 9 application, regardless to the libraries using it.

If your application can benefit from using newer features *directly*, multi-targeting is a good idea then.

**Comment by JKorf** [comment]
Alright, thanks for the answer. I was a bit confused bit this snippet as found on https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting:

![image](https://github.com/user-attachments/assets/7fb1f3d3-5396-42e1-967b-5cd163627f12)

**Reply by huoyaoyuan** [reply]
Binding redirect is a legacy concept for old frameworks. It doesn't apply to .NET 5+.
